athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/propertygraph/rowwriters/EdgeRowWriter.java

```diff
@@ -77,33 +82,33 @@ public static void writeRowTemplate(RowWriterBuilder rowWriterBuilder, Field fie
                 rowWriterBuilder.withExtractor(field.getName(),
                         (VarCharExtractor) (Object context, NullableVarCharHolder value) -> {
                             Map<Object, Object> obj = (Map<Object, Object>) context;
+                            String fieldName = getFieldName(field.getName().toLowerCase().trim(), obj);
```

Comment:
maybe I'm missing something, but how does a case sensitive match happen if I'm always lower casing the fieldName? Can you add some unit tests to show how this works?

Reply:
Yup, this is error prone.

I would suggest that the author of this patch look into DDBTypeUtils.java:contextAsMap and also see how it gets used and follow that pattern instead:
https://github.com/awslabs/aws-athena-query-federation/blob/master/athena-dynamodb/src/main/java/com/amazonaws/athena/connectors/dynamodb/util/DDBTypeUtils.java#L368

Reply:
The challenge here is to pick the correct key from underlying data store because Glue converts everything into lower case. So whether you store FirstName, FiRstName or firstname Glue always sends FirstName. I need to pick correct key match from Neptune properties. The final match of properties anyways has to be converted into Lowercase

Reply:
If you know a way to read properties from Glue Table the way it was stored let me know, I couldn't find any reference for it.

Reply:
This is error prone because it requires every site that performs a lookup into the obj map:
1) Cast `Object obj` into `Map<Object, Object> map`.

2) To lowercase the `fieldName` field before passing it into this function.

3) Pass in the resolved `fieldName` back into the `map` to look up the value.

If you look at the `DDBTypeUtils.java:contextAsMap` way of doing things instead, the map itself becomes case insensitive so you never have to worry about the calling sites lowercasing before passing in.

Using the `DDBTypeUtils.java:contextAsMap` pattern it becomes:

1) Convert `Object obj` into `Map<String, Object> map`  using `contextAsMap(obj, caseInsensitive)`. This returns you a Map that is either case sensitive or not depending on the parameter.

2) Simply lookup `fieldName` from `map` without lowercasing or anything.

This means that if you did things like `DDBTypeUtils.java:contextAsMap` it just works without the developer ever having to think about manually lowecasing `fieldName` when `enableCaseinsensitivematch` is set to true (this just replaces the straight cast with a function call instead of the cast).

Think about this from the aspect of maintainability too, people that add code to this function won't have to think about lowercasing the `fieldName` before looking things up if you just had a case insensitive map to begin with.

The pattern using `DDBTypeUtils.java:contextAsMap` is the most direct expression of "case insensitive lookup" rather than what is currently being done in this pull request, which is: "resolve the original cased key from a lowercased key by searching the map keys and then look up the value from the map using the resolved original cased key."

So even while your assumption that the fieldNames are always going to be lowercased is correct, it is strictly simpler and less error prone to never have to make that assumption or think about it to begin with.

Reply:
Makes sense, so if I understand it correctly. A call to obj.get  would return value irrespective of key being uppercase or lowercase or mixedcase. Will make changes and test

Reply:
Yeah, that's the idea.

Reply:
updated code to use contextAsMap.

Neptune doesn't return simple key value map. For keys ID, IN and OUT the returned key is of type struct